AppController/djinn.rb

```diff
@@ -5730,14 +5875,30 @@ def get_all_stats(secret)
           if app_name != "none"
             total_reqs, reqs_enqueued, collection_time = get_haproxy_stats(app_name)
             # Create the apps hash with useful information containing HAProxy stats.
-            all_stats["apps"][app_name] = {
-              "language" => @app_info_map[app_name]["language"].tr('^A-Za-z', ''),
-              "appservers" => @app_info_map[app_name]["appengine"].length,
-              "http" => @app_info_map[app_name]["nginx"],
-              "https" => @app_info_map[app_name]["nginx_https"],
-              "total_reqs" => total_reqs,
-              "reqs_enqueued" => reqs_enqueued
-            }
+            tries = MAX_RETRIES
+            begin
+              all_stats["apps"][app_name] = {
+                  "language" => @app_info_map[app_name]["language"].tr('^A-Za-z', ''),
+                  "appservers" => @app_info_map[app_name]["appengine"].length,
+                  "http" => @app_info_map[app_name]["nginx"],
+                  "https" => @app_info_map[app_name]["nginx_https"],
+                  "total_reqs" => total_reqs,
+                  "reqs_enqueued" => reqs_enqueued
+              }
+            rescue => exception
+              backtrace = exception.backtrace.join("\n")
+              message = "Unforseen exception: #{exception} \nBacktrace: #{backtrace}"
+              
+              if tries > 0
+                Djinn.log_warn("Try: #{tries} Unforseen exception: #{exception} \nBacktrace: #{backtrace} \n")
+                Kernel.sleep(SMALL_WAIT)
+                tries -= 1
+                retry
+              else
+                @state = message
+                HelperFunctions.log_and_crash(@state, WAIT_TO_CRASH)
```

Comment:
Do we want to crash the whole controller here? It seems like we should just send an error message back to the client.